analyzers/src/SonarAnalyzer.Common/Rules/UseShortCircuitingOperatorBase.cs

```diff
@@ -39,35 +38,35 @@ protected static bool IsBool(SyntaxNode node, SemanticModel semanticModel)
         }
     }
 
-    public abstract class UseShortCircuitingOperatorBase<TLanguageKindEnum, TBinaryExpression> : UseShortCircuitingOperatorBase
-        where TLanguageKindEnum : struct
+    public abstract class UseShortCircuitingOperatorBase<TSyntaxKind, TBinaryExpression> : UseShortCircuitingOperatorBase<TSyntaxKind>
+        where TSyntaxKind : struct
         where TBinaryExpression : SyntaxNode
     {
-        protected sealed override void Initialize(SonarAnalysisContext context)
-        {
-            context.RegisterNodeAction(
-                GeneratedCodeRecognizer,
-                c =>
+        protected sealed override void Initialize(SonarAnalysisContext context) =>
+            context.RegisterNodeAction(Language.GeneratedCodeRecognizer, c =>
                 {
-                    var node = (TBinaryExpression)c.Node;
-
-                    if (GetOperands(node).All(o => IsBool(o, c.SemanticModel)))
+                    if (c.Node is TBinaryExpression node
+                        && Language.Syntax.BinaryExpressionLeft(node) is { } left
+                        && Language.Syntax.BinaryExpressionRight(node) is { } right
+                        && IsBool(left, c.SemanticModel)
+                        && IsBool(right, c.SemanticModel))
                     {
+                        var extractText = c.SemanticModel.GetConstantValue(right) is { HasValue: true }
+                            || c.SemanticModel.GetSymbolInfo(right).Symbol is ILocalSymbol or IFieldSymbol or IPropertySymbol or IParameterSymbol
```

Comment:
Ok for `ILocalSymbol`, `IFieldSymbol` and `IParameterSymbol`.
As for `IPropertySymbol`: I guess the assumption is that the getter "plays fair", and doesn't produce side effects, such as raising exceptions or validating preconditions.
Is this the rationale behind adding `IPropertySymbol` here?

Reply:
IPropertySymbol getters are already presumed to play nice, e.g. with pattern matching. As such, we can assume the same here.